dask/dataframe/io/tests/test_hdf.py

```diff
@@ -415,6 +415,14 @@ def test_read_hdf(data, compare):
         assert (sorted(dd.read_hdf(fn, '/data', mode='r').dask) ==
                 sorted(dd.read_hdf(fn, '/data', mode='r').dask))
 
+    with tmpfile('h5') as fn:
+        sorted_data = data.sort_index()
+        sorted_data.to_hdf(fn, '/data', format='table')
+        a = dd.read_hdf(fn, '/data', chunksize=2, sorted_index=True, mode='r')
+        assert a.npartitions == 2
+
+        compare(a.compute(), sorted_data)
```

Comment:
Nice test.  Thank you for this

Reply:
Not directly related but the previous test check that:
`sorted(dd.read_hdf(fn, '/data', mode='r').dask) == sorted(dd.read_hdf(fn, '/data', mode='r').dask)`

What is the intended behaviour ?

Reply:
That the names of the tasks are deterministic.  This is useful for a few reasons, one is to avoid duplicate work.